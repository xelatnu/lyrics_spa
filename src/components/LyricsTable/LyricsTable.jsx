import React from 'react';
import TableHeader from '../LyricsTableHeader/LyricsTableHeader';
import TableRow from '../lyricsTableRow/LyricsTableRow';
// import './lyricsTable.css';

const LyricsTable = ({ lyricsData }) => (
  <table className="table table-dark table-hover">
    <TableHeader />
    {lyricsData.map(({
      song, artist, album, ...restItem
    }, index) => (
      <TableRow
          // eslint-disable-next-line react/no-array-index-key
        key={index}
        song={song}
        songLink={restItem['song-link']}
        artist={artist}
        artistLink={restItem['artist-link']}
        album={album}
      />
    ))}
  </table>
);
export default LyricsTable;
