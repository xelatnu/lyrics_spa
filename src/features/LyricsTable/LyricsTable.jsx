/* eslint-disable react/prop-types */
import React from 'react';

const LyricsTable = ({ lyrics }) => (
  <table data-testid="lyricsTable" className="table table-dark table-hover">
    <thead>
      <tr>
        <th scope="col">#</th>
        <th scope="col">Song</th>
        <th scope="col">Artist</th>
        <th scope="col">Album</th>
      </tr>
    </thead>
    {lyrics?.map(({
      song, artist, album, ...restItem
    }, index) => {
      console.log(song);

      return (
        <tbody>
          <tr>
            <th scope="row">{index + 1}</th>
            <td><a className="lyrics_link link-light" target="_blank" href={restItem['song-link']} rel="noreferrer">{song}</a></td>
            <td><a className="lyrics_link link-light " target="_blank" href={restItem['artist-link']} rel="noreferrer">{artist}</a></td>
            <td><a className="lyrics_link link-light " target="_blank" href={restItem['album-link']} rel="noreferrer">{album}</a></td>
          </tr>
        </tbody>
      );
    })}
  </table>
);

export default LyricsTable;
