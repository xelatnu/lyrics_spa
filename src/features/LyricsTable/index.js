import LyricsTable from './LyricsTable';

export default LyricsTable;
