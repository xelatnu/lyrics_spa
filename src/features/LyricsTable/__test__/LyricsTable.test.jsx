import React from 'react';
import renderWithProviders from '../../../utils/test-utils';
import LyricsTable from '../LyricsTable';

describe('Render <LyricsTable/>', () => {
  it('should render lyrics table header', () => {
    const { asFragment } = renderWithProviders(<LyricsTable />);
    expect(asFragment(<LyricsTable />)).toMatchSnapshot();
  });
});
