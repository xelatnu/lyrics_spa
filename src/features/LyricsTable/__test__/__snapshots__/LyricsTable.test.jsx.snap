// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <LyricsTable/> should render lyrics table header 1`] = `
<DocumentFragment>
  <table
    class="table table-dark table-hover"
    data-testid="lyricsTable"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          #
        </th>
        <th
          scope="col"
        >
          Song
        </th>
        <th
          scope="col"
        >
          Artist
        </th>
        <th
          scope="col"
        >
          Album
        </th>
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;
